src/pages/logging.tsx

```diff
@@ -0,0 +1,312 @@
+      }
+    };
+
+    fetchData();
+  }, []);
+
+  const transformData = (data: CallData[]): number[] => {
+    let callsByInterval: number[] = [];
+    if (filterBy === 'day') {
+      callsByInterval = new Array(24).fill(0);
+      data.forEach((call: CallData) => {
+        const startTime = new Date(call.startTime);
+        const hour = startTime.getHours();
+        callsByInterval[hour]++;
+      });
+    } else if (filterBy === 'week') {
+      callsByInterval = new Array(7).fill(0);
+      data.forEach((call: CallData) => {
+        const startTime = new Date(call.startTime);
+        const dayOfWeek = startTime.getDay();
+        callsByInterval[dayOfWeek]++;
+      });
+    } else if (filterBy === 'month') {
+      callsByInterval = new Array(31).fill(0);
+      data.forEach((call: CallData) => {
+        const startTime = new Date(call.startTime);
+        const dayOfMonth = startTime.getDate();
+        callsByInterval[dayOfMonth - 1]++;
+      });
+    }
+    return callsByInterval;
+  };
+
+  const transformDurationData = (data: CallData[]): number[] => {
+    let durationsByInterval: number[] = [];
+    if (filterBy === 'day') {
+      durationsByInterval = new Array(24).fill(0);
+      data.forEach((call: CallData) => {
+        const startTime = new Date(call.startTime);
+        const hour = startTime.getHours();
+        durationsByInterval[hour] += call.duration;
+      });
+    } else if (filterBy === 'week') {
+      durationsByInterval = new Array(7).fill(0);
+      data.forEach((call: CallData) => {
+        const startTime = new Date(call.startTime);
+        const dayOfWeek = startTime.getDay();
+        durationsByInterval[dayOfWeek] += call.duration;
+      });
+    } else if (filterBy === 'month') {
+      durationsByInterval = new Array(31).fill(0);
+      data.forEach((call: CallData) => {
+        const startTime = new Date(call.startTime);
+        const dayOfMonth = startTime.getDate();
+        durationsByInterval[dayOfMonth - 1] += call.duration;
+      });
+    }
+    return durationsByInterval;
+  };
+
+  const calculateAverageDurations = (callData: CallData[], durationsByInterval: number[]): number[] => {
+    const callCounts = transformData(callData);
+    return durationsByInterval.map((totalDuration, index) =>
+      callCounts[index] > 0 ? totalDuration / callCounts[index] : 0
+    );
+  };
+
+  const transformedData: number[] = transformData(callData);
+  const transformedDurations: number[] = transformDurationData(callData);
+  const avgDurationsByInterval: number[] = calculateAverageDurations(callData, transformedDurations);
+
+  useEffect(() => {
+    if (callChartDataRef.current === null) return;
+    const ctx = callChartDataRef.current.getContext('2d') as CanvasRenderingContext2D;
+
+    let labels: string[] = [];
+    let data: number[] = [];
+    let yAxesMax = 0;
+
+    if (filterBy === 'day') {
+      yAxesMax = 2500; // Maximum value on y-axes for 'day' filter
```

Comment:
Currently the yAxesMax values are set arbitrarily. Maybe we can set them dynamically, like 1.25 * (max value in transformedData)?